File: waimai/api/src/main/java/club/yunzhi/log/entity/DishesOrder.java
package club.yunzhi.log.entity;


import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;

import javax.persistence.*;
import java.sql.Timestamp;

@Entity
@Getter
@Setter
public class DishesOrder {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @OneToOne
    private Dishes dishes;

    private Integer count;

    private float price;

    @CreationTimestamp
    private Timestamp createTime;

}

File: web/src/entity/dishesOrder.ts
/**
 * 专业
 * author: liguowen
 */
import {Restaurant} from './restaurant';
import {Dishes} from './dishes';

export class DishesOrder {
  /**
   * id
   */
  id?: number;

  dishes?: Dishes;

  count?: number;

  createTime?: number;

  price?: number;

  constructor(data = {} as {
    id?: number,
    dishes?: Dishes,
    count?: number,
    createTime?: number,
    price?: number
  }) {
    this.id = data.id;
    this.dishes = data.dishes;
    this.count = data.count;
    this.createTime = data.createTime;
    this.price = data.price;
  }
}
